util/update-repo.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+if test "$#" -ne 1; then
+    echo "$0 software-stack | data | compute | io | app-interact" 1>&2
+    exit 1
+fi
+
+# Only proceed if repository is clean.
+# Long live StackExchange: https://unix.stackexchange.com/a/155077/227174
+output=$(git status --untracked-files=no --porcelain)
+if test ! -z "$output"; then
+    echo "Repository is not clean. Run 'git status' to see uncommitted changes." 1>&2
+    echo "Please commit / clean your changes before update." 1>&2
```

Comment:
This moves the task of committing their changes to the students. I like this better than running
```bash
git add .
git commit -m 'some message'
```
within the script, despite the extra time it takes.